src/property.hpp

```diff
@@ -62,7 +62,22 @@ struct Property {
     using IsPrimary = util::TaggedBool<class IsPrimaryTag>;
     using IsIndexed = util::TaggedBool<class IsIndexedTag>;
 
+    // The internal column name used in the Realm file.
     std::string name;
+
+    // An alias for the internal column names in the Real file. Bindings can use to expose a different name,
+    // e.g. to map between different naming conventions.
+    //
+    // If no alias is defined a property `alias` is equal to its `name`.
+    //
+    // ObjectStore will ensure that no conflicts occur between persisted properties and the alias, so
+    // an alias is just as unique an identifier as the property name in the file.
+    //
+    // In order to respect aliases bindings should use `ObjectSchema::property_for_alias()`
+    // and XXX for read fields defined by an alias.
+    // If alias are allowed in queries, an XXX must be parsed to XXX. It is up to bindings to
+    // keep track of this.
+    std::string alias;
```

Comment:
So only one alias?

Reply:
Yes, I don't think there is any use case for having multiple?

Reply:
Do any of the bindings already call this an alias? It'd be nice to avoid introducing yet another totally different concept for how column names and property names relate to each other.

Reply:
See https://realmio.slack.com/archives/CA41SMVPV/p1554116647000200

Both Java and .NET support this, but neither call it alias. Instead, they talk about mapping from public names to internal names using slightly different terminology. Java uses `@RealmField("name")` and .NET `[MapTo("name)]`.

Since they both are about mapping public names to internal names, I should probably reword this. However, since the `Property` works on internal names already I probably prefer the word `publicName` as it feels more descriptive. Thoughts?

Reply:
`name` (a.k.k `internal_name` but renaming it would be really annoying) +`public_name` seems fine to me. If we stick with the constraint that there needs to still be a field named `name` which means roughly what it currently does to avoid creating a ton of extra work for ourselves I don't think there's a *great* option.